### PR TITLE
feat: execute tests in headless mode

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

Prevents browser windows to be opened when executing tests. However, headless mode is always deactivated when running the test in debug mode.

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,11 @@ This deploys demo at http://localhost:8080
 
 To run Integration Tests, execute `mvn verify -Pit,production`.
 
+Tests run by default in `headless` mode, to avoid browser windows to be opened for every test.
+This behaviour is always disabled when running the tests in debug mode in the IDE
+or when running maven with the `-Dmaven.failsafe.debug` sytem property.
```
Comment (tltv) [inline README.md]:
```suggestion
or when running maven with the `-Dmaven.failsafe.debug` system property.
```

File: src/test/java/org/vaadin/addons/mygroup/AbstractViewTest.java [hunk 2]
```diff
@@ -78,8 +96,8 @@ protected WebElement getRootElement() {
      * identified by {@code themeClass}. If the the is not found, JUnit
      * assert will fail the test case.
      *
-     * @param element       web element to check for the theme
-     * @param themeClass    theme class (such as {@code Lumo.class}
+     * @param element    web element to check for the theme
+     * @param themeClass theme class (such as {@code Lumo.class}
```
Comment (tltv) [inline src/test/java/org/vaadin/addons/mygroup/AbstractViewTest.java]:
```suggestion
     * @param themeClass theme class (such as {@code Lumo.class})
```

**Review by tltv — CHANGES_REQUESTED** [review]

**Review by tltv — APPROVED** [review]

**Review by czp13 — APPROVED** [review]

**Comment by czp13** [comment]
Probably would be good to test this manually to see if tests are working in all the ways (debug, property, without those).

**Comment by mcollovati** [comment]
I did all the mentioned tests locally. It's quite easy to do it if you want to double check all is working as expected

**Comment by mcollovati** [comment]
@samie @mstahv, do you mind taking a look at this?

**Review by mstahv — APPROVED** [review]
Looks good to me!

**Pull request merged**